### PR TITLE
Join profinet multicast group (linux)

### PR DESCRIPTION
**Pull request by olbjo**
Some linux solutions needs this addition to receive
Profinet multicasts. DCP Identity all do not work
without it.

Change-Id: Ia7fb34d1c23e1ebcacda2dd76964c2ae488584c3

### CODE REVIEW

File: src/ports/linux/pnal_eth.c [hunk 1]
```diff
@@ -142,6 +143,24 @@ pnal_eth_handle_t * pnal_eth_init (
    sll.sll_protocol = htons (linux_receive_type);
    bind (handle->socket, (struct sockaddr *)&sll, sizeof (sll));
 
+   /* Join profinet multicast group */
+   mreq.mr_ifindex = ifindex;
+   mreq.mr_type = PACKET_HOST | PACKET_MR_MULTICAST;
+   mreq.mr_alen = ETH_ALEN;
+   const uint8_t pn_mcast_addr[ETH_ALEN] = {0x01, 0x0e, 0xcf, 0x00, 0x00, 0x00};
```
Comment (pyhys) [inline src/ports/linux/pnal_eth.c:150]: Vi brukar lägga definitioner i början av funktionen

**Review by pyhys — COMMENTED** [review]
Ser bra ut. En minimal justering.

File: src/ports/linux/pnal_eth.c (hunk 1 above)
Comment (olbjo) [reply src/ports/linux/pnal_eth.c]: OK

**Review by pyhys — APPROVED** [review]
Sweet!

File: src/ports/linux/pnal_eth.c [hunk 2]
```diff
@@ -142,6 +144,23 @@ pnal_eth_handle_t * pnal_eth_init (
...
+
+   if (
+      setsockopt (
+         handle->socket,
+         SOL_PACKET,
+         PACKET_ADD_MEMBERSHIP,
+         &mreq,
+         sizeof (mreq)) != 0)
+   {
+      printf ("Failed to join Profinet multicast group\n");
```
Comment (hefloryd) [inline src/ports/linux/pnal_eth.c:161]: This should probably be a log message of some sort. Should the error be propagated to the caller?

Comment (olbjo) [reply src/ports/linux/pnal_eth.c]: I added the log message as a PNAL message. All pnal files share the same tag.
Regarding the message I just wanted to keep this problem visible and then address it if it is a problem on other systems. 

Return codes for other ioctls or options are not checked in this function. It has worked without this additional socket option with previous linux systems we have been using and I do not think we need to propagate the error to user. 

I do not know why additional socket option is needed in this particular case. Just tried make it work.

**Pull request merged**